### PR TITLE
[release/6.0] Onboard new inter-branch merge flow [Workflow]

### PR DESCRIPTION
**Pull request by f-alizada**
### Context
Onboarding new merge flow based on the github actions

### Changes Made
Added github workflow for the branch.

### References
Documentation https://github.com/dotnet/arcade/blob/main/Documentation/Maestro/New-Inter-Branch-Merge-Approach.md
One pager: https://github.com/dotnet/dnceng/blob/main/Documentation/OnePagers/github-action-inter-branch-merge.md

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]

File: .github/workflows/inter-branch-merge-flow.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    branches:
+      - release/**
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  Merge:
+    uses: dotnet/arcade/.github/workflows/inter-branch-merge-base.yml@main
```
Comment (ericstj) [inline .github/workflows/inter-branch-merge-flow.yml:13]: Is it normal to use `main` in a release branch?  I would expect each release branch to use the respective release branch of arcade (and ideally at a matching commit to that arcade which is ingested in the repo).  @mmitche

**Comment by ericstj** [comment]
@f-alizada @mmitche can you help explain to us what this does.  I see a lot of mention of `inter-branch merges` but I'm not sure what is meant by that.  Can you give us an example?

File: .github/workflows/inter-branch-merge-flow.yml (hunk 1 above)
Comment (mmitche) [reply .github/workflows/inter-branch-merge-flow.yml:13]: We use main in arcade for publishing and other common automation (including this automation in its current form). The scripting for inter-branch merging has changed minimally over the years and isn't release specific in any way.

**Comment by mmitche** [comment]
@ericstj This is the automation that merges between branches in GitHub (e.g. release/6.0->release/7.0 in some repos). It generates these PRs: https://github.com/dotnet/runtime/pull/104330

This automation is moving from being triggered by old Maestro, to being triggered by GitHub Actions

**Comment by carlossanlop** [comment]
@mmitche so does this look good to merge?

**Review by mmitche — APPROVED** [review]

**Comment by mmitche** [comment]
Yep good to go.

**Pull request merged**